src/dmd/iasmgcc.d

```diff
@@ -284,27 +285,22 @@ Ldone:
 public Statement gccAsmSemantic(GccAsmStatement s, Scope *sc)
 {
     //printf("GccAsmStatement.semantic()\n");
-    scope diagnosticReporter = new StderrDiagnosticReporter(global.params.useDeprecated);
-    scope p = new Parser!ASTCodegen(sc._module, ";", false, diagnosticReporter);
 
     // Make a safe copy of the token list before parsing.
-    Token *toklist = null;
-    Token **ptoklist = &toklist;
+    TokenRange toklist = s.tokens;
+    auto p = cast(Parser!ASTCodegen)toklist.getLexer;
```

Comment:
Don't call a lexer a `p`.

Reply:
Call it `lex` or `lexer`.

Reply:
It's being used as a parser because it has to call parseExpression later on.    I don't love this design but couldn't come up with a simpler alternative